feugy/tabulous
# feat(web): displays peers selections

### :book: What's in there?

Now that each player has its own color, we can start displaying their respective selections.

[peer-selection.webm](https://user-images.githubusercontent.com/186268/201526286-91b13b5b-ae3b-4358-99d6-51ff9be3d9fe.webm)

Please note that, while meshes selected by someone else can not be selected, they are still interactive (will be prevented in another piece of work).

Are included here:

- fix(web): can't type 'h' in invite dialogue (considered as shortcut)
- feat(web): displays and keeps in sync peers selection
- feat(web): sends active selection to peers
- feat(web): adds player colors to indicators
- refactor(web): renders mesh edges when selecting them. Does not alter color player.
- refactor(server): uses more contrasted colors.
- chore(cli): migrates code from istanbul to c8 coverage tool

### :test_tube: How to test?

1. Make a new game that has several seats
2. Invite other players
   > each players gets one of the 8 colors randomly assigned
3. Select some meshes before other player connect
   > your selection appears with your color, and is more visible
4. Let other peers connect
   > they will see your selection, and won't be able to select these meshes
5. Let other peers select some meshes
   > you should see their selection, with their own colors, and these meshes are not selectable for you
6. Let another peer disconnect
   > their selection is cleared.
